repos/fdbt-site/cypress_tests/cypress/support/steps.ts

```diff
@@ -351,3 +353,42 @@ export const completeMyFaresOtherProductsPages = (): void => {
         clickElementByText('Back');
     }
 };
+
+export const editServicesOtherProductsPage = (): void => {
+    for (let i = 0; i < 3; i++) {
+        clickRandomElementInTable('govuk-table__body', 'product-link');
+        getElementById('product-name').should('not.be.empty');
+        getElementById('product-status').should('not.be.empty');
+        getElementById('fare-type').should('not.be.empty');
+        clickElementById('Services-link');
+        randomlyChooseAndSelectServices();
+        continueButtonClick();
+        clickElementById('Passenger-type-link');
+        randomlyDetermineUserType();
+        clickElementById('Start-date-link');
+        clearDates();
+        completeProductDateInformationPage();
+        clickElementById('End-date-link');
+        clearDates();
+        completeProductDateInformationPage();
+        clickElementById('Time-restriction-link');
```

Comment:
I think we need to add some validation to these steps, for example change the thing you are editing to something different, and then once it redirects you back you can validate that the new value is shown. Currently, the edit functionality could be broken and we wouldn't know it with these tests.

If we do go down that route, it might be best to split them into multiple different tests, rather than one large one. That way, if a test fails then it wont make everything else fail, and we will be able to easily figure out which part is broken.

Reply:
Managed to add it for product name but the rest is a bit hard to check because i reused functions that randomise it

Reply:
did it for name, services, start and end date - rest to be resolved in another ticket